pytorch_tools/utils/tiles.py

```diff
@@ -0,0 +1,188 @@
+"""
+Implementation of tile-based inference allowing to predict huge images that does not fit into GPU memory entirely
+in a sliding-window fashion and merging prediction mask back to full-resolution.
+
+Hacked together by @zakajd & @bonlime
+
+Reference:
+    https://github.com/BloodAxe/pytorch-toolbelt/blob/develop/pytorch_toolbelt/inference/tiles.py
+    https://github.com/xinntao/Real-ESRGAN/blob/master/realesrgan/utils.py
+    https://github.com/victorca25/iNNfer
+"""
+import itertools
+from typing import Callable, Optional, List
+
+import torch
+from torch.nn import functional as F
+
+
+def compute_pyramid_patch_weight_loss(width: int, height: int, depth: Optional[int] = None) -> torch.Tensor:
+    """Compute a weight matrix that assigns bigger weight on pixels in center and
+    less weight to pixels on image boundary. This weight matrix then used for merging
+    individual tile predictions and helps dealing with prediction artifacts on tile boundaries.
+
+    Args:
+        width: Tile width
+        height: Tile height
+        depth: Tile depth (used in 3d case).
+
+    Returns:
+        Weight matrix of shape (1, width, height, [depth])
+
+    """
+    Dcx = torch.linspace(-width * 0.5 + 0.5, width * 0.5 - 0.5, width).square()
+    Dcy = torch.linspace(-height * 0.5 + 0.5, height * 0.5 - 0.5, height).square()
+    Dcz = 0
+    if depth is not None:
+        Dcz = torch.linspace(-depth * 0.5 + 0.5, depth * 0.5 - 0.5, depth).square()
+        Dcz = Dcz[None, None, :]
+
+    # eucl distance to center
+    Dc = (Dcx[:, None, None] + Dcy[None, :, None] + Dcz).sqrt()
+
+    De_x = torch.linspace(0.5, width - 0.5, width).abs()
+    De_x = torch.minimum(De_x, De_x.flip(dims=(0,)))
+
+    De_y = torch.linspace(0.5, height - 0.5, height).abs()
+    De_y = torch.minimum(De_y, De_y.flip(dims=(0,)))
+
+    De_z = torch.tensor(torch.inf)
```

Comment:
При попытке запустить код упало с ошибкой `AttributeError: module 'torch' has no attribute 'inf'`. Решается заменой на:
```suggestion
    De_z = torch.tensor(float('inf'))
```